pkg/crc/services/dns/template.go

```diff
@@ -35,6 +37,21 @@ const (
 	zoneFileAddressTemplate = `; Name Server Information
 _etcd-server-ssl._tcp.{{ .ClusterName}}.{{ .BaseDomain }}. IN SRV 10 10 2380 etcd-0.{{ .ClusterName }}.{{ .BaseDomain }}.
 *.{{ .BaseDomain }}.                IN  A     {{ .IP }}
+`
+
+	dnsmasqConfTemplate = `user=root
+bind-interfaces
+expand-hosts
+log-queries
+srv-host=_etcd-server-ssl._tcp.{{ .ClusterName}}.{{ .BaseDomain }},etcd-0.{{ .ClusterName}}.{{ .BaseDomain }},2380,10
+local=/{{ .ClusterName}}.{{ .BaseDomain }}/
+domain={{ .ClusterName}}.{{ .BaseDomain }}
+address=/{{ .AppsDomain }}/{{ .IP }}
+address=/etcd-0.{{ .ClusterName}}.{{ .BaseDomain }}/{{ .IP }}
+address=/api.{{ .ClusterName}}.{{ .BaseDomain }}/{{ .IP }}
```

Comment:
Wouldn't `address=/.{{ .ClusterName}}.{{ .BaseDomain }}/{{ .IP }}` be enough to also catch `etcd-0`, `api`, etc

or even `address=/.{{ .BaseDomain }}/{{ .IP }}` to catch all